### PR TITLE
ENG-3857 Retrieval of tags and digests from Docker registries

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

File: src/commands/generate-cr.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  DOCKER_ORGANIZATION_PROPERTY,
+  DOCKER_REGISTRY_PROPERTY
+} from '../services/config-service'
+import {
+  DEFAULT_DOCKER_REGISTRY,
+  DockerService
+} from '../services/docker-service'
+
+export default class GenerateCr extends Command {
+  static description = 'Generate CR for a bundle project'
```
Comment (cecchisandrone) [inline src/commands/generate-cr.ts]: Generate the Entando Custom Resource (CR) for a bundle project

File: src/commands/generate-cr.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  static examples = [
+    '<%= config.bin %> <%= command.id %>',
+    '<%= config.bin %> <%= command.id %> --image=my-org/my-bundle',
+    '<%= config.bin %> <%= command.id %> --image=my-registry/my-org/my-bundle'
+  ]
+
+  static flags = {
+    image: Flags.string({
+      char: 'i',
+      description: 'Name of the bundle Docker image'
```
Comment (cecchisandrone) [inline src/commands/generate-cr.ts]: add the accepted formats

**Review by raxkaynan — APPROVED** [review]

**Comment by zonia3000** [comment]
@cecchisandrone I set the parallelism for digests retrieval to 6. With that and the added progressbar I think it looks usable also with library/mysql (~32 seconds)

File: src/commands/generate-cr.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+  static flags = {
+    image: Flags.string({
+      char: 'i',
+      description:
+        'Name of the bundle Docker image with the format ' + VALID_BUNDLE_FORMAT
+    }),
+    digest: Flags.boolean({
+      char: 'd',
+      description: 'Include the digests of Docker images'
```
Comment (cecchisandrone) [inline src/commands/generate-cr.ts]: Include Docker images digests

File: src/commands/generate-cr.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      }
+
+      const configService = new ConfigService()
+      const dockerOrganization = configService.getProperty(
+        DOCKER_ORGANIZATION_PROPERTY
+      )
+
+      if (!dockerOrganization) {
+        this.error(
+          'Docker organization not configured for the project. Bundle image must be published to generate CR.'
```
Comment (cecchisandrone) [inline src/commands/generate-cr.ts]: Docker organization not configured for the project. Bundle image must be published to generate the CR.

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**